include/event2/http_struct.h

```diff
@@ -142,6 +142,12 @@ struct {
 	 */
 	void (*on_complete_cb)(struct evhttp_request *, void *);
 	void *on_complete_cb_arg;
+
+	/*
+	 * Free callback - called just before the request is freed.
+	 */
+	void (*on_free_cb)(struct evhttp_request *, void *);
```

Comment:
This is an ABI break. The soname needs to be bumped.

Reply:
Actually using `evhttp_request` directly is not recommended, but yes it is.

And there is a comment in this file:

```
Data structures for http.  Using these structures may hurt forward
compatibility with later versions of Libevent: be careful!
```

But nobody reads headers